### DISCUSSION TITLE
How to get dns name from the spefific tags filter in version3?

### DISCUSSION THREAD

**Discussion by madhusudhan1234** [aws/aws-sdk-ruby#2600] [Q&A]
Hi Community, 

While doing some Ruby and RoR upgrade work I had to upgrade the aws-sdk gem. Previous implementation to get the certain dns name for the deployment in version 1 is:

```Ruby
  # Initialization 
  AWS.config(access_key_id: 'aws_access_key_id',
              secret_access_key: 'aws_secret_access_key',
              region: 'region')

```
```Ruby 
web_instances = AWS.ec2.instances
                   .filter('tag:name1', 'value1')
                   .filter('tag:name2', 'value2')
                   .filter('tag:name3', 'value3')
                   .sort{|i1, i2| i1.launch_time <=> i2.launch_time} # oldest first
                   .map{|instance| instance.dns_name}
``` 

Now I tried different approaches to get the same result of dns name in version 3. I am still not able to do, Can you please help on on this? 

Thanks in advance.

**Comment by mullermp** [comment]
I don't know much about V1, but I think the service clients are similar to the Resource models that V2/V3 use. I wouldn't recommend using those anymore.

I think what you should be doing is using [describe_instances](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#describe_instances-instance_method) with your filters.

**Comment by madhusudhan1234** [comment]
Hi @mullermp, Thanks for the reply.

As per your suggestion I implemented in this way.

```Ruby 
ec2 = Aws::EC2::Client.new(region: 'us-west-2', credentials: Aws::Credentials.new(config['aws_access_key_id'], config['aws_secret_access_key']))

instances = ec2.describe_instances(filters: [
    { name: 'tag:name1', values:['value1']},
    { name: 'tag:name2', values:['value1', 'value2']},
    { name: 'tag:name3', values:['value3']}
  ])
```

And got the following result in `puts instances.to_h`.
```Ruby
{:reservations=>[]}
```

Did I miss anything in the implementation? Thanks in advance.

**Reply by mullermp** [reply]
I think that's correct, but I'm not sure why your list is empty. Perhaps the filtering is too strict.
